metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala

```diff
@@ -149,16 +148,9 @@ final class InteractiveSemanticdbs(
     }
   }
 
-  def getBuildTarget(
```

Comment:
This is still needed - when looking into files in the `./metals/readonly` we try to infer the build target (and the right presentation compiler) based on where the file comes from. For example if we opened a file in project A and go to definition opened another file for example `Predef.scala` we will associate `Predef.scala` with project A.

I think that actually we can move it to `BuildTargets` class and create`inferBuildTarget` method. This we could use in the `Compilers` class instead of always doing`.orElse(interactiveSemanticdbs.flatMap(_.getBuildTarget(path)))` 

We would need to also move `toFileOnDisk`, `readonlyToSource`, `didDefinition`, and `inferBuildTarget` but that should not be an issue.

Reply:
I deleted it because `BuildTargets` already has `inferBuildTarget` method. 
It's implemented a little bit different way but looks similar. I tested that `references` in `./metals/readonly` still works in this PR

Reply:
Och, yeah there is also `inferBuildTarget` method but it is much more taxing (creates new class loader each time :O). However, it seems that `inferBuildTarget` here does not make much sense. We do save everything to database and it will stay longer than the in memory `readonlyToSource`. I think it's a good idea to remove it, it's probably a leftover.

We should also remove then `readonlyToSource`, which does not make any sense anymore and `toFileOnDisk`, which we can just use the normal underlying `toFileOnDisk` in `WorkspaceSymbolProvider`.

It would also be good to cache the `URLClassLoader` that gets created there, it may actually create a memory leak, but it can be done in a separate PR.

Reply:
> We should also remove then readonlyToSource

:+1:  Done

> It would also be good to cache the URLClassLoader that gets created there, it may actually create a memory leak, but it can be done in a separate PR.

Agree about doing this in separate PR. Will do.